libjob_queue/src/lsf_driver.c

```diff
@@ -347,35 +347,43 @@ stringlist_type * lsf_driver_alloc_cmd(lsf_driver_type * driver ,
   */
 
   {
-    stringlist_type * excludes = stringlist_alloc_new();
-    char * select = NULL;
-    if (driver->exclude_hosts != NULL) {
+    stringlist_type * select_list = stringlist_alloc_new();
+    if (driver->exclude_hosts != NULL && stringlist_get_size(driver->exclude_hosts) > 0) {
       for (int i = 0; i < stringlist_get_size(driver->exclude_hosts); i++) {
         char * exclude_host = util_alloc_sprintf("hname!='%s'", stringlist_iget(driver->exclude_hosts, i));
-        stringlist_append_owned_ref(excludes, exclude_host);
+        stringlist_append_owned_ref(select_list, exclude_host);
       }
-      const char * excludes_string = stringlist_alloc_joined_string(excludes, " && ");
-      select = util_alloc_sprintf("select[%s]", excludes_string);
     }
 
-    stringlist_type * qrr = stringlist_alloc_new();
-    if (select) {
-      stringlist_append_copy(qrr, select);
-      util_free(select);
-    }
-
-    if (driver->resource_request != NULL)
-      stringlist_append_copy(qrr, driver->resource_request);
+    char * excludes_string = NULL;
+    char * req = "";
 
```

Comment:
There are four cases here.  `driver->resource_request` can be empty/NULL, but is typically `select[cs && x86_64Linux]`, and `driver->exclude_hosts`  can be empty, or a list of hostnames (e.g. `[node1, node2]`).

In the end we are interested in a string `"select[cs && whatever && hname!=node1 && hname!=node2]"`.

What is done here, is to remove the `']'` from `driver->resource_request` and concatenate `stringlist_alloc_joined_string(select_list, " && ")` on the end of it, if it exists.

Or to create `"select[hname!=node1 && hname!=node2]"` if no resource_request.